### DISCUSSION TITLE
Inconsistent plugins options type identifiers

### DISCUSSION THREAD

**Discussion by jean-michelet** [fastify/fastify#5591] [General]
> EDIT: no need for deprecation, I just change the type (see https://github.com/fastify/fastify/discussions/5591#discussioncomment-10167908)

Core plugin option type identifiers are not consistent across the project. Some are prefixed with `Fastify`, others are not. I think we should choose a convention, deprecate the types that don't respect this convention and remove them in the next major.

Examples :
```ts
// prefixed
import helmet, { FastifyHelmetOptions } from "@fastify/helmet";
import fastifyCors, { FastifyCorsOptions } from "@fastify/cors";

// not prefixed
import fastifyMysql, { MySQLOptions } from "@fastify/mysql";
import sensible, { SensibleOptions } from "@fastify/sensible";
import fastifyUnderPressure, { UnderPressureOptions } from "@fastify/under-pressure";
```

I suggest prefixing with `Fastify`, as some plugins wrap a library and the option name can be confusing (e.g. `MySQLOptions`, altough there is usually a namespace to add context `fastifyMysql.MySQLOptions`). Also, I suggest to explicitly type the plugin configuration in the [official demo](https://github.com/fastify/demo) for educational purpose, and imo it will look better with consistent type identifiers.

I can investigate the plugins and make the changes oc, I create this discussion to link it each time I push a PR for context.

**Comment by jean-michelet** [comment]
I should make a todo list here, but I already pushed un bunch of PR.
Waiting for comments if any.

fastify-mysql: https://github.com/fastify/fastify-mysql/pull/144
fastify-sensible: https://github.com/fastify/fastify-sensible/pull/162
fastify/under-pressure: https://github.com/fastify/under-pressure/pull/229

Todo:
- [ ] fastify-autoload
... searching for more...

**Reply by mcollina** [reply]
there is no need for deprecations. Just outright make the changes.

**Reply by jean-michelet** [reply]
Ok
